modules/auxiliary/scanner/http/allegro_rompager_misfortune_cookie.rb

```diff
@@ -27,38 +27,140 @@ def initialize(info = {})
       ],
       'References' => [
         ['CVE', '2014-9222'],
-        ['URL', 'http://mis.fortunecook.ie']
+        ['URL', 'http://mis.fortunecook.ie'],
+        ['URL', 'http://mis.fortunecook.ie/misfortune-cookie-suspected-vulnerable.pdf'], # list of likely vulnerable devices
+        ['URL', 'http://mis.fortunecook.ie/too-many-cooks-exploiting-tr069_tal-oppenheim_31c3.pdf'] # 31C3 presentation with POC
       ],
       'DisclosureDate' => 'Dec 17 2014',
       'License' => MSF_LICENSE
     ))
 
-    register_options([
-      OptString.new('TARGETURI', [true, 'Path to fingerprint RomPager from', '/Allegro'])
-    ], self.class)
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'URI to test', '/'])
+      ], Exploit::Remote::HttpClient
+    )
   end
 
-  def check_host(ip)
-    res = send_request_cgi('uri' => normalize_uri(target_uri.path.to_s), 'method' => 'GET')
+  def check_host(_ip)
+    begin
+      test_misfortune
+    ensure
+      disconnect
+    end
+  end
+
+  def run_host(ip)
+    case check_host(ip)
+    when Exploit::CheckCode::Appears
+      print_good("#{peer} is vulnerable")
+    when Exploit::CheckCode::Detected
+      print_good("#{peer} uses a vulnerable version")
+    else
+      vprint_status("#{peer} is not vulnerable")
+    end
+  end
+
+  def check_response_fingerprint(res, fallback_status)
     fp = http_fingerprint(response: res)
-    if /RomPager\/(?<version>[\d\.]+)$/ =~ fp
+    if /RomPager\/(?<version>[\d\.]+)/ =~ fp
+      vprint_status("#{peer} is RomPager #{version}")
       if Gem::Version.new(version) < Gem::Version.new('4.34')
+        return Exploit::CheckCode::Detected
```

Comment:
As you indicated in a subsequent comment, oftentimes a short-circuit return is much better.  Explicit returns are generally only frowned upon when they are not required, for example at the end of a method or simple conditionals.  That isn't the case here.